libs/execution_context/ebpf_program.c

```diff
@@ -140,16 +146,22 @@ _ebpf_program_program_info_provider_changed(
     size_t total_helper_count = 0;
     ebpf_helper_function_addresses_t* total_helper_function_addresses = NULL;
 
+    // Detach
     if (provider_data == NULL) {
-        ebpf_rundown_protection_wait(&program->rundown_protection);
-        program->context_create = NULL;
-        program->context_destroy = NULL;
+        // The NMR model is async, but the only Windows run-down protection API available is a blocking API, so the
+        // following call will block until all using threads are complete. This should be fixed in the future.
+        // Issue: https://github.com/microsoft/ebpf-for-windows/issues/1854
+        ebpf_rundown_protection_wait(&program->rundown.protection);
 
         // Extension is detaching. Program will get invalidated.
         program->info_extension_provider_binding_context = NULL;
-        program->info_extension_provider_data = NULL;
+        program->rundown.info_extension_provider_data = NULL;
         goto Exit;
-    } else {
+    } else
+    // Attach
+    {
```

Comment:
nit: could we move the comment after this opening brace and it in-line with the ```else```?